bom/application/pom.xml

```diff
@@ -4430,7 +4430,7 @@
             </dependency>
             <dependency>
                 <groupId>org.graalvm.sdk</groupId>
-                <artifactId>graal-sdk</artifactId>
+                <artifactId>nativeimage</artifactId>
                 <version>${graal-sdk.version}</version>
             </dependency>
```

Comment:
I think we need to keep `graal-sdk` in the BOM for now to avoid breaking the whole extension ecosystem.

Reply:
If we do so, then the extensions might keep pulling in graal-sdk and thus triggering issues like https://github.com/quarkusio/quarkus/issues/39322 and https://github.com/quarkusio/quarkus/issues/39350. no?

Reply:
OK, I suppose extension developers can live with the break.

Reply:
I'll push this PR to 3.9 so please add it to the 3.9 migration guide.

Reply:
Note that we need this backported to 3.8 as well if we want backward compatibility with Mandrel 23.0 (see https://github.com/quarkusio/quarkus/issues/39322)

Reply:
Yeah that's why I'm not very comfortable with us breaking all the extensions out there...

Not sure if it's a decision we should take hidden in an issue.

Reply:
Breaking almost every [other] extension again (dep changed _to_ graal-sdk not that long ago), mid-stream this time, to keep compatibility with something () they seem less likely to hit than (older Mandrel builder image... which you dont actually test with anymore?)...doesnt seem a no-brainer.

The older quarkus boms set provided scope for the older svm dependency. The newer boms dont set it for graal-sdk. Probably it shouldnt be set, but as it does seem to be the scope that should be used....could the old dep be left in place, but 'provided' scope added? Setting provided scope in my extension resolved the build failure I saw with older Mandrel 23.0.3.0 based JDK17 builder image.

(speaking as someone whose extension will be broken by the removal)

Reply:
> older Mandrel builder image... which you dont actually test with anymore?

FTR that's not true. [Quarkus core is being tested with Mandrel 23.0 on a weekly base](https://github.com/graalvm/mandrel/blob/6d0a3fc45652f00cb92a4be68441e27cc43621b6/.github/workflows/weekly.yml#L111-L122).

Reply:
Based on the above discussion I added `graal-sdk` back in the BOM file. If we end up dropping it I suggest we do it in a separate PR after discussing this more broadly.

Reply:
> could the new dep still be added to the 3.8 bom?

@gemmellr yes the new dep is most likely going to be added to the 3.8 bom.

Reply:
> > older Mandrel builder image... which you dont actually test with anymore?
> 
> FTR that's not true. [Quarkus core is being tested with Mandrel 23.0 on a weekly base](https://github.com/graalvm/mandrel/blob/6d0a3fc45652f00cb92a4be68441e27cc43621b6/.github/workflows/weekly.yml#L111-L122).

That particular 'you dont test' comment was aimed at the _Quarkus_ side of things, where it does seem to be true?

Is it tested (from the Mandrel side) _before every single Quarkus / Platform release_ happens? If not, then seems more like it might eventually be noticed it was broken and released already and maybe fixed later. I wouldnt really count that personally.

Reply:
@gsmet FYI the bacward compatibility issue with Mandrel 23.0 is now handled in https://github.com/quarkusio/quarkus/pull/39397.

> Is it tested (from the Mandrel side) before every single Quarkus / Platform release happens?

It is tested weekly and we fix issues as they show up. As we are testing with Quarkus `main` the issues are typically resolved before every Quarkus release, and thus before every Platform release as well. Note however that we only test Quarkus core, so issues that might appear on the extensions side are not in our radar.

Reply:
> It is tested weekly and we fix issues as they show up. As we are testing with Quarkus `main` the issues are typically resolved before every Quarkus release, and thus before every Platform release as well. Note however that we only test Quarkus core, so issues that might appear on the extensions side are not in our radar.

Right, so _main_  is tested, weekly, but none of the actual e.g _3.8_ or _3.9_ releases going forward are themselves directly tested with it, as those release branches now only use the newer version builders in their testing. Testing main as an aproximation will likely catch most general issues, but due to subsequent new additions on main and/or selective backports, might not catch issues with the releases, which it still seems are not tested directly before or after release.

I'm not saying this is a bad thing if its unclear, I was simply noting earlier (when I didnt know about the Mandrel-side testing at all) that compatibility with the old builder version seemed a questionable reason to justify immediately breaking most other extensions, when that compatibility isnt even actually being tested directly with those releases. I still think that, but its no longer an issue now with the latest changes thankfully.